terraform-beyond-layer0/example-2/main.tf

```diff
@@ -0,0 +1,91 @@
+#
+# ./example2/main.tf
+#
+
+provider "layer0" {
+  endpoint        = "${var.endpoint}"
+  token           = "${var.token}"
+  skip_ssl_verify = true
+}
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region     = "${var.region}"
+}
+
+# create one layer0 environment for each map value defined in variable 'environments'
+resource "layer0_environment" "demo" {
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
+}
+
+# create one loadbalancer for each map value defined in variable 'environments'
+resource "layer0_load_balancer" "guestbook" {
+  count = "${length(var.environments)}"
+
+  name        = "${lookup(var.environments, count.index)}_guestbook_lb"
```

Comment:
I would keep the `name` the same value. Seems redundant to need to do 
```
l0 loadbalancer get dev:dev_guestbook_db
l0 loadbalancer get prod:dev_guestbook_db
``` 
When, if they had the same names, they could do:
```
l0 loadbalancer get dev:guestbook-lb
l0 loadbalancer get prod:guestbook-lb
```